File: python/leetcode/LEETCODE_7.py
class Solution:
    def reverse(self, x: int) -> int:
        rev: int = 0
        bound: float = math.pow(2, 31)
            
        if x >= 0:
            rev = int(str(x)[::-1])
        else:
            rev = -int(str(x)[1:][::-1])            
        
        if rev < -bound or rev >= bound:
            return 0
        else:
            return rev

File: java/src/leetcode/LEETCODE_7.java
package leetcode;

class LEETCODE_7 {
    public int reverse(int x) {
        long rev = 0;
        double bound = Math.pow(2, 31);
        
        boolean isNegative = x < 0;
        long num = Math.abs((long) x);
        
        while (num >= 10) {
            long mod = num % (long) 10;
            num = num / (long) 10;
            rev = 10 * rev + mod;
        }
        rev = 10 * rev + num;
        
        if (rev < -bound || rev >= bound) {
            return 0;
        } else if (isNegative) {
            return (int)-rev;
        } else {
            return (int) rev;
        }
    }
}
